coreAPI/src/main/java/net/java/games/input/DefaultControllerEnvironment.java

```diff
@@ -153,6 +112,55 @@ public DefaultControllerEnvironment() {
         }
         return ret;
     }
+
+	/**
+	 * This method will populate the controllers list and is meant to be called manually
+	 * in order to refresh the list of available controllers.
+	 */
+	public void getNewControllers(){
+		controllers = new ArrayList<>();
+		AccessController.doPrivileged((PrivilegedAction<Void>) () -> scanControllers());
+		//Check the properties for specified controller classes
+		String pluginClasses = getPrivilegedProperty("jinput.plugins", "") + " " + getPrivilegedProperty("net.java.games.input.plugins", "");
+		if(!getPrivilegedProperty("jinput.useDefaultPlugin", "true").toLowerCase().trim().equals("false") && !getPrivilegedProperty("net.java.games.input.useDefaultPlugin", "true").toLowerCase().trim().equals("false")) {
+			String osName = getPrivilegedProperty("os.name", "").trim();
+			if(osName.equals("Linux")) {
+				pluginClasses = pluginClasses + " net.java.games.input.LinuxEnvironmentPlugin";
+			} else if(osName.equals("Mac OS X")) {
+				pluginClasses = pluginClasses + " net.java.games.input.OSXEnvironmentPlugin";
+			} else  if(osName.equals("Windows XP") || osName.equals("Windows Vista") || osName.equals("Windows 7") || osName.equals("Windows 8") || osName.equals("Windows 8.1") || osName.equals("Windows 10")) {
+				pluginClasses = pluginClasses + " net.java.games.input.DirectAndRawInputEnvironmentPlugin";
+			} else if(osName.equals("Windows 98") || osName.equals("Windows 2000")) {
+				pluginClasses = pluginClasses + " net.java.games.input.DirectInputEnvironmentPlugin";
+			} else if (osName.startsWith("Windows")) {
+				log.warning("Found unknown Windows version: " + osName);
+				log.warning("Attempting to use default windows plug-in.");
+				pluginClasses = pluginClasses + " net.java.games.input.DirectAndRawInputEnvironmentPlugin";
+			} else {
+				log.warning("Trying to use default plugin, OS name " + osName +" not recognised");
+			}
+		}
+
+		StringTokenizer pluginClassTok = new StringTokenizer(pluginClasses, " \t\n\r\f,;:");
+		while(pluginClassTok.hasMoreTokens()) {
+			String className = pluginClassTok.nextToken();
+			try {
+				if(!loadedPluginNames.contains(className)) {
+					log.fine("Loading: " + className);
+					Class<?> ceClass = Class.forName(className);
+					ControllerEnvironment ce = (ControllerEnvironment) ceClass.getDeclaredConstructor().newInstance();
+					if(ce.isSupported()) {
+						addControllers(ce.getControllers());
```

Comment:
As the plugins are not expecting a second call, we should either be requesting any changes to the devices list from the plugin, or calling a clean up method before we get the controllers again, as the plugins may be opening resources multiple times otherwise. Alternatively, the plugins should be updated to handle the resources correctly in their own getControllers methods prior to returning the list if this is not the first call.